frontend/src/scenes/insights/utils.tsx

```diff
@@ -433,3 +434,9 @@ export function insightUrlForEvent(event: Pick<EventType, 'event' | 'properties'
 
     return query ? urls.insightNew(undefined, undefined, query) : undefined
 }
+
+export function isQueryTooLarge(query: Node<Record<string, any>>): boolean {
+    // Chrome has a 2MB limit for the HASH params, limit ours at 1MB
+    const queryLength = JSON.stringify(query).length
+    return queryLength > 1024 * 1024
```

Comment:
might be wrong here but you might need something [more complex](https://stackoverflow.com/a/68791382/5540417) to check number of bytes used by a JS string? :)

https://stackoverflow.com/a/52254083/5540417